src/cow-react/modules/limitOrders/hooks/useLimitOrdersPriceImpactParams.ts

```diff
@@ -0,0 +1,28 @@
+import { useSafeMemoObject } from '@cow/common/hooks/useSafeMemo'
+import { useLimitOrdersTradeState } from '@cow/modules/limitOrders/hooks/useLimitOrdersTradeState'
+import type { PriceImpactParams } from '@src/custom/hooks/usePriceImpact'
+import { useMemo } from 'react'
+
+export function useLimitOrdersPriceImpactParams(): PriceImpactParams {
+  const { inputCurrencyAmount, outputCurrencyAmount } = useLimitOrdersTradeState()
+
+  const abTrade = useSafeMemoObject({
```

Comment:
why useMemo on this one, just to add it as a dependency to useMemo in the other one?

Reply:
Unfortunately, `useSafeMemoObject` can be efficient only with flat objects, it's a reason why I can't pass the whole `PriceImpactParams` into it.